src/MySqlConnector/Serialization/MySqlSession.cs

```diff
@@ -526,47 +526,81 @@ private void VerifyConnected()
 
 		private async Task InitSslAsync(ProtocolCapabilities serverCapabilities, ConnectionSettings cs, IOBehavior ioBehavior, CancellationToken cancellationToken)
 		{
-			X509CertificateCollection clientCertificates = null;
+			X509CertificateCollection CertificateFiles = null;
 			if (cs.CertificateFile != null)
 			{
 				try
 				{
-					var certificate = new X509Certificate2(cs.CertificateFile, cs.CertificatePassword);
-					clientCertificates = new X509CertificateCollection {certificate};
+					X509Certificate2 certificate;
+					if (cs.CertificatePassword == null)
+						certificate = new X509Certificate2(cs.CertificateFile);
+					else
+						certificate = new X509Certificate2(cs.CertificateFile, cs.CertificatePassword);
```

Comment:
Thanks for the CoreFX link. I wanted to make sure importing PEM encoded certs would work too and usually those aren't password protected. I'll change it back though since it runs through the same codepath.